classes/UpgradeTools/ModuleAdapter.php

```diff
@@ -129,17 +130,37 @@ public function listModulesToUpgrade(array $modulesFromAddons)
         }
 
         foreach (scandir($dir) as $module_name) {
-            if (is_file($dir . DIRECTORY_SEPARATOR . $module_name)) {
+            // We don't update autoupgrade module
+            if ($module_name === 'autoupgrade') {
                 continue;
             }
-
+            // We have a file modules/mymodule
+            if (is_file($dir . $module_name)) {
```

Comment:
No `DIRECTORY_SEPARATOR` needed (previous code used it)?

Reply:
As defined in `classes/UpgradeContainer.php`, it already contains a `DIRECTORY_SEPARATOR` at the end.